src/main/java/io/leonis/zosma/game/Identity.java

```diff
@@ -12,10 +12,10 @@
   /**
    * Supplies an {@link Identity}.
    */
-  interface Supplier {
+  interface Supplier<I extends Identity> {
     /**
      * @return The identity of the supplier.
      */
-    Identity getIdentity();
+    I getIdentity();
```

Comment:
I don't think this supplier should have an generic. It's only used for equality right?